solutions/instances/main.tf

```diff
@@ -7,7 +7,15 @@ locals {
   # tflint-ignore: terraform_unused_declarations
   validate_log_analysis_provision = var.enable_at_event_routing_to_log_analysis && var.log_analysis_provision == false ? tobool("log_analysis_provision can't be false if enable_at_event_routing_to_log_analysis is true") : true
   # tflint-ignore: terraform_unused_declarations
+  validate_log_analysis_archive_provision = var.log_analysis_enable_archive && var.log_analysis_provision == false ? tobool("log_analysis_provision can't be false if log_analysis_enable_archive is true") : true
```

Comment:
I dont think you need this validation - just dont take any archive action if `log_analysis_provision` is set to false